internal/runtime/safecall.go

```diff
@@ -0,0 +1,69 @@
+package runtime
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+)
+
+// SafeCall safely calls a function using reflection. It handles potential
+// panics by recovering and returning an error. The function `fn` is expected
+// to be a function, and `args` are the arguments to pass to that function.
+// It returns the result of the function call (if any) and an error if one
+// occurred during the call or if a panic was recovered.
+func SafeCall(fn any, args ...any) (result any, err error) {
+	// Ensure fn is a function
+	fnValue := reflect.ValueOf(fn)
+	fnType := fnValue.Type()
+	if fnValue.Kind() != reflect.Func {
+		return nil, errors.New("fn is not a function")
+	}
+
+	// Defer a function to handle panics
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("recovered from panic: %v", r)
```

Comment:
Maybe you could add a sentinel error, and include it in this error wrapper, so you will be able to catch a panic.

```go
var ErrPanicRecovery = errors.New("recovered from a panic")
```

```suggestion
			err = fmt.Errorf("%w: %v", ErrPanicRecovery, r)
```

It's a suggestion

Reply:
Good suggestion, I integrate it